data/trie/leafNode.go

```diff
@@ -111,20 +111,39 @@ func (ln *leafNode) hashNode() ([]byte, error) {
 	return encodeNodeAndGetHash(ln)
 }
 
-func (ln *leafNode) commit(force bool, _ byte, _ uint, _ data.DBWriteCacher, targetDb data.DBWriteCacher) error {
+func (ln *leafNode) commitDirty(_ byte, _ uint, _ data.DBWriteCacher, targetDb data.DBWriteCacher) error {
 	err := ln.isEmptyOrNil()
 	if err != nil {
 		return fmt.Errorf("commit error %w", err)
 	}
 
-	shouldNotCommit := !ln.dirty && !force
-	if shouldNotCommit {
+	if !ln.dirty {
 		return nil
 	}
 
 	ln.dirty = false
 	return encodeNodeAndCommitToDB(ln, targetDb)
 }
+func (ln *leafNode) commitCheckpoint(_ data.DBWriteCacher, targetDb data.DBWriteCacher) error {
+	err := ln.isEmptyOrNil()
+	if err != nil {
+		return fmt.Errorf("commit checkpoint error %w", err)
+	}
+
+	//TODO add early return if should not commit to checkpoint db and set commited flag to true
```

Comment:
implement both TODOs?

Reply:
Will be implemented in the following PR, because it requires a lot of code changes.

Reply:
ok